gson/src/main/java/com/google/gson/stream/JsonReader.java

```diff
@@ -907,7 +907,7 @@ public double nextDouble() throws IOException {
     peeked = PEEKED_BUFFERED;
     double result = Double.parseDouble(peekedString); // don't catch this NumberFormatException.
     if (!lenient && (Double.isNaN(result) || Double.isInfinite(result))) {
-      throw new MalformedJsonException(
```

Comment:
Seems like it is indeed bad JSON since JSON forbids NaN and infinities. 
The number may actually be in correct format.

Reply:
Though then the question is why `NumberFormatException` is thrown in the first place when the number cannot be parsed as double, int or long since it is malformed JSON in these cases as well.